bindings/haskell/haskell-src/OpenDAL.hs

```diff
@@ -63,6 +67,28 @@ data ErrorCode
 data OpenDALError = OpenDALError {errorCode :: ErrorCode, message :: String}
   deriving (Eq, Show)
 
+data EntryMode = File | Dir | Unknown deriving (Eq, Show)
+
+data BytesContentRange = BytesContentRange
+  { bcrStart :: Integer
+  , bcrEnd :: Integer
+  , bcrTotal :: Integer
+  }
+  deriving (Eq, Show)
+
+data Metadata = Metadata
+  { mMode :: EntryMode
+  , mCacheControl :: Maybe String
+  , mContentDisposition :: Maybe String
+  , mContentLength :: Integer
+  , mContentMD5 :: Maybe String
+  , mContentRange :: Maybe BytesContentRange
+  , mContentType :: Maybe String
+  , mETag :: Maybe String
+  , mLastModified :: Maybe POSIXTime
```

Comment:
Does `POSIXTime` here contains time zone?

Reply:
`PosixTime` is the nominal time since 1970-01-01 00:00 UTC.

I convert `DateTime<Utc>` to unix timestamp according to `timestamp()` function, and then transfer it to haskell. I think `chrono` crate has dealed with time zone.

Reply:
I expect to have more precise time than just a timestamp (e.g. in nanoseconds, or at least milliseconds). And I prefer to store as a native datetime struct instead of a timestamp (if haskell has one).

However, it's okay to support in the next pull request.

Reply:
We can use `UTCTime` in haskell. We should export `DateTime` to a string with `RFC 3339` format, and then parse it in haskell.